#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_gcc41.dpatch by Guido Trotter <ultrotter@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix compilation errors with gcc 4.1

@DPATCH@
diff -urNad freej-0.8.1/lib/flash/graphic16.h /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/graphic16.h
--- freej-0.8.1/lib/flash/graphic16.h	2005-08-06 22:35:03.000000000 +0200
+++ /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/graphic16.h	2006-06-21 15:25:33.000000000 +0200
@@ -24,7 +24,7 @@
 
 class GraphicDevice16: public GraphicDevice {
 private:
-	long GraphicDevice16::allocColor(Color color);
+	long allocColor(Color color);
 
 public:
 	GraphicDevice16(FlashDisplay *fd);
diff -urNad freej-0.8.1/lib/flash/graphic24.h /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/graphic24.h
--- freej-0.8.1/lib/flash/graphic24.h	2005-08-06 22:35:04.000000000 +0200
+++ /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/graphic24.h	2006-06-21 15:25:33.000000000 +0200
@@ -24,7 +24,7 @@
 
 class GraphicDevice24: public GraphicDevice {
 private:
-	long GraphicDevice24::allocColor(Color color);
+	long allocColor(Color color);
 
 public:
 	GraphicDevice24(FlashDisplay *fd);
diff -urNad freej-0.8.1/lib/flash/graphic32.h /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/graphic32.h
--- freej-0.8.1/lib/flash/graphic32.h	2005-08-06 22:35:03.000000000 +0200
+++ /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/graphic32.h	2006-06-21 15:25:33.000000000 +0200
@@ -24,7 +24,7 @@
 
 class GraphicDevice32: public GraphicDevice {
 private:
-	long GraphicDevice32::allocColor(Color color);
+	long allocColor(Color color);
 
 public:
 	GraphicDevice32(FlashDisplay *fd);
diff -urNad freej-0.8.1/lib/flash/matrix.h /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/matrix.h
--- freej-0.8.1/lib/flash/matrix.h	2005-08-06 22:35:04.000000000 +0200
+++ /tmp/dpep.h1nW5s/freej-0.8.1/lib/flash/matrix.h	2006-06-21 15:25:33.000000000 +0200
@@ -33,13 +33,13 @@
 #endif
 
 	inline
-	long Matrix::getX(long x, long y)
+	long getX(long x, long y)
 	{
 		return (long) (x*a+y*b+tx);
 	};
 
 	inline
-	long Matrix::getY(long x, long y)
+	long getY(long x, long y)
 	{
 		return (long) (x*c+y*d+ty);
 	};
diff -urNad freej-0.8.1/src/include/blitter.h /tmp/dpep.h1nW5s/freej-0.8.1/src/include/blitter.h
--- freej-0.8.1/src/include/blitter.h	2005-08-06 22:34:35.000000000 +0200
+++ /tmp/dpep.h1nW5s/freej-0.8.1/src/include/blitter.h	2006-06-21 15:25:33.000000000 +0200
@@ -115,7 +115,7 @@
   bool fade_value(int step, int val); ///< fade to a new blit value
   bool pulse_value(int step, int val); ///< pulse it to a value and come back
   bool set_kernel(short *krn); /// set the convolution kernel
-  bool Blitter::set_colorkey(int x,int y);
+  bool set_colorkey(int x,int y);
 
   bool set_zoom(double x, double y);
   bool set_rotate(double angle);
diff -urNad freej-0.8.1/src/include/linklist.h /tmp/dpep.h1nW5s/freej-0.8.1/src/include/linklist.h
--- freej-0.8.1/src/include/linklist.h	2005-08-06 22:34:35.000000000 +0200
+++ /tmp/dpep.h1nW5s/freej-0.8.1/src/include/linklist.h	2006-06-21 15:25:33.000000000 +0200
@@ -50,7 +50,7 @@
   Entry *search(char *name);
   int *completion(char *needle);
 
-  Entry *Linklist::selected();
+  Entry *selected();
 
   Entry *operator[](int pos) { return pick(pos); };
 
